### ISSUE TITLE
Disappearing cursor

### ISSUE THREAD

**Issue by dnkl** [swaywm/wlroots#3189]
wlroots: https://github.com/swaywm/wlroots/commit/20d404a091fa92f56be4825bbaf89e4027fda154
sway: ec5da0ca5bad6a433f727499d68ac1352397f5aa

Not sure if this is a problem with sway, or not. But, I'm now using sway on multiple systems, with the same software setup.

Everything is working fine on all systems except one, where the mouse cursor keeps disappearing. It seems its position is correctly tracked though, because e.g. hover effects in browsers work, focus-follows-mouse works and so, on. It's just that the cursor itself is invisible.

The cursor is usually invisible right from the start, before any windows have been opened. Then, if I move it between windows, it will sometimes become visible. When that happens, it stays visible until moved to another window.

The sway debug log doesn't contain any obvious error messages, except these:
```
2019-02-07 21:22:15 - [sway/input/cursor.c:1132] denying request to set cursor from unfocused client
```
and
```
glamor: No eglstream capable devices found
```

The main difference between the non-working system and the working ones are that the non-working one has an nvidia card (with nouveau), while the others are intel based.

**Comment by Munksgaard** [comment]
> Is this a regression? I don't remember having this problem on 1.5.x, and I already had an HDMI monitor.

It certainly is for me.

**Comment by Ulexus** [comment]
Also seems to be a regression for me.  I lost my external monitor for a couple months, and now that I have it back, I am seeing this issue.  The only changes have been software updates: it was working just fine a few months ago, and now when I plug in the external monitor, have the monitors go to sleep and then wake back up, the cursor is invisible on the laptop's builtin display but visible on the external display.  Single (internal) display:  no problems.

Arch Linux, AMD Renoir integrated GPU, sway 1.6.1.

**Comment by Ulexus** [comment]
Indeed:  in my case, downgrading to the 1.6.0 sway package seems to prevent the problem.

**Comment by edrex** [comment]
This condition is triggered for me mid-session after launching certain programs (eg, the Arch AUR obsidian-appimage package).

One interesting bit I've noticed is that keeping the `wdisplays` program running keeps cursors visible, although the `failed to render cursor buffer` errors keep scrolling by.

**Comment by RX14** [comment]
Judging by the timing of when this issue appeared for me, and that some cursor-related items appeared in the changelog, could the issue have appeared in 0.14.1 and not 0.14.0?

**Comment by jackinloadup** [comment]
I've experiencing something as well. I can keep the mouse visible when wdisplays is open but otherwise it is hidden. Oddly it's only on one display. The primary one DP-1 while DP-2 will display the mouse fine regardless. I've only noticed the behavior after waking from swaylock and maybe twice in the last 7 days.

The workaround I've come up with is to pull out and put back in the monitor cable.

**Comment by sykhro** [comment]
> The workaround I've come up with is to pull out and put back in the monitor cable.

Interestingly, that does nothing for me.

**Comment by skligys** [comment]
Does wdisplay disable hardware mouse cursor?  It solves the problem for me but mouse movement becomes noticeably more sluggish.

**Comment by jackinloadup** [comment]
I don't know about wdisplay disabling the hardware cursor but i do know if wdisplay is open while some videos are playing on screen it can cause a performance drop. I'm guessing because of the video preview aspect of wdisplay, which is pretty slick. Though you can disable the preview once wdisplays is open there is a hamburger menu which allows you to toggle "Show Screen Contents" and another one "Overlay Screen Names". Disabling both and throwing wdisplays to a hidden workspace or scratchpad might be a viable temporary workaround. Granted I haven't tested to see if disabling "Show Screen Contents" actually improves performance, only an assumption.

**Comment by apvodney** [comment]
I've had this issue, switching to a TTY without sway on it and back seems to bring the cursor back, nixos 21.05, sway 1.6.1
